frontend/src/components/pki-syncs/forms/PkiSyncOptionsFields/PkiSyncOptionsFields.tsx

```diff
@@ -87,6 +94,66 @@ export const PkiSyncOptionsFields = () => {
           </FormControl>
         )}
       />
+
+      <Controller
+        control={control}
+        name="syncOptions.certificateNameSchema"
+        render={({ field: { value, onChange }, fieldState: { error } }) => (
+          <FormControl
+            tooltipClassName="max-w-md"
+            tooltipText={
+              <div className="flex flex-col gap-3">
+                <span>
+                  When a certificate is synced, values will be injected into the certificate name
+                  schema before it reaches the destination. This is useful for organization.
+                </span>
+
+                <div className="flex flex-col">
+                  <span>Available placeholders:</span>
+                  <ul className="list-disc pl-4 text-sm">
+                    <li>
+                      <code>{"{{certificateId}}"}</code> - The unique ID of the certificate
+                    </li>
+                    <li>
+                      <code>{"{{environment}}"}</code> - The environment which the certificate is in
+                      (e.g. dev, staging, prod)
+                    </li>
+                  </ul>
+                </div>
+                {syncOption?.forbiddenCharacters && syncOption.forbiddenCharacters.length > 0 && (
+                  <div className="flex flex-col">
+                    <span className="text-yellow">
+                      Character restrictions for {syncOption.name}:
+                    </span>
+                    <div className="text-xs text-bunker-300">
+                      The following characters are not allowed:{" "}
+                      {syncOption.forbiddenCharacters.split("").join(" ")}
+                    </div>
+                    {syncOption.allowedCharacterPattern && (
+                      <div className="mt-1 text-xs text-bunker-300">
+                        Only alphanumeric characters and hyphens are allowed (a-z, A-Z, 0-9, -)
+                      </div>
+                    )}
```

Comment:
we would be displaying a label here that's based on the value of `allowedCharacterPattern` right? Curious, how would we parse the pattern into text?